app/forms/import_duplicate_form.rb

```diff
@@ -49,6 +49,26 @@ def discard_pending_changes!
   end
 
   def keep_both_changes!
-    object.apply_pending_changes_to_new_record! if can_keep_both?
+    if can_keep_both?
```

Comment:
If the user chooses to accept the pending changes, I think we would also need to apply this logic.